explorer/interpreter/resolve_names.cpp

```diff
@@ -277,13 +293,20 @@ auto NameResolver::AddExposedNames(const Declaration& declaration,
 auto NameResolver::ResolveNames(Expression& expression,
                                 const StaticScope& enclosing_scope)
     -> ErrorOr<std::optional<ValueNodeView>> {
+  if (trace_stream_->is_enabled()) {
+  }
   return RunWithExtraStack(
       [&]() { return ResolveNamesImpl(expression, enclosing_scope); });
 }
 
 auto NameResolver::ResolveNamesImpl(Expression& expression,
                                     const StaticScope& enclosing_scope)
     -> ErrorOr<std::optional<ValueNodeView>> {
+  if (trace_stream_->is_enabled()) {
+    *trace_stream_ << "** resolving expr `";
+    expression.Print(trace_stream_->stream());
+    *trace_stream_ << "` (" << expression.source_loc() << ")\n";
+  }
```

Comment:
Tracing every expression we enter and leave seems like it's going to be very verbose. Perhaps we can just trace statements and declarations, and not expressions and where clauses and patterns and the like?

Reply:
Yes, it is getting verbose when tracing enter/leave for every expression, where clause and pattern.
Tracing only statements and declarations looks like a better choice to me as well.